controllers/admin/AdminAccessController.php

```diff
@@ -223,4 +226,30 @@ private function getChildrenTab(array &$tabs, $id_parent = 0)
 
         return $children;
     }
+
+    /**
+     * @param int $tab_id
+     * @param string $permission
+     *
+     * @return bool
+     *
+     * @throws Exception
+     */
+    private function isAllowedToEditPermissions($tab_id, $permission)
+    {
+        if ($tab_id === -1) {
+            return false;
+        }
+
+        $currentProfile = (int) ContextCore::getContext()->employee->id_profile;
+        $slug = Access::findSlugByIdTab($tab_id);
+
+        foreach ((array) Access::getAuthorizationFromLegacy($permission) as $legacyAuthorization) {
+            if (!Access::isGranted($slug . $legacyAuthorization, $currentProfile)) {
+                return false;
+            }
+        }
```

Comment:
Could you please add some comments on what you're checking here?

Reply:
I don't know what this function does under the hood.

Reply:
to be more precise, I don't know for sure what does do Access::isGranted() so I've done some assumptions about it and it works as expected 😇

Reply:
I added the comments by myself. We could not approve stuff we don't understand.